templates/go/message.go

```diff
@@ -7,8 +7,31 @@ const messageTpl = `
 	{{ if .MessageRules.GetSkip }}
 		// skipping validation for {{ $f.Name }}
 	{{ else }}
-		if v, ok := interface{}({{ accessor . }}).(interface{ Validate() error }); ok {
-			if err := v.Validate(); err != nil {
+		if v, ok := interface{}({{ accessor . }}).(interface{ ValidateWithMask(*field_mask.FieldMask) error }); m.maskHas(mask, "{{ $f.Name }}") && ok {
+			// update the mask to remove the outer level
+			if mask != nil {
+				paths := []string{}
+				for _, path := range mask.GetPaths() {
+					parts := strings.SplitN(path, ".", 2)[1:]
+					if len(parts) > 0 {
+						paths = append(paths, strings.Join(parts, "."))
```

Comment:
This seems buggy since it is affecting all paths, not just the ones relevant to the inner message. Consider a FieldMask that looks like this:
```proto
paths: "a.c.d"
paths: "b.c.e"
```
Then when looking at field "a", we'd expect to only validate it's "c.d" field, but since we also chop off the "b" from the second path, we'd also end up validating "a.c.e".

Reply:
hm, good catch!  :)

Reply:
So I tried doing this by traversing `pgs.Field` to the enclosing `Message()` and its `Parent()`, but it appears that this doesn't work as I hoped... IIUC, this traverses based on the way messages are written in the file (i.e., parsed at compile or build-time) rather than the way they're setup at runtime (e.g., KitchenSinkMessage contains ComplexTestMsg `val`, which contains `another` ComplexTestMsg, which contains an `int_const`). The PGS library lets me go from `int_const` to `ComplexTestMsg`, but I can't go to `another` or `val` via this hierarchy.

I'm new to protoc-gen-star, but is this the correct understanding?

Reply:
Ok, nevermind. Much easier solution. Pushed.